src/textinput/styles/textinput.css

```diff
@@ -0,0 +1,21 @@
+@import '../../styles/variables.css';
+
+.textinput {
+	font: var(--base-font);
+}
+
+.textinput input {
```

Comment:
can you please drop the usage of tag names and give a class name to both the `input` and the `span` then you can drop the `.textinput` nesting as it is not required.